subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,24 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+//printk("%u %u\n", sys_get_be32(NET_TCP_HDR(pkt)->seq), context->tcp->send_ack);
+
+	/* Peer sent us packet we've already seen. Apparently, our ack was lost. */
+	if (sys_get_be32(NET_TCP_HDR(pkt)->seq) < context->tcp->send_ack) {
+printk("Received already seen pkt - ACKing\n");
+
+	set_appdata_values(pkt, IPPROTO_TCP);
```

Comment:
> can't we just send an ACK that is using currently active ack values?

And yep, we can't do this (if taken literally) - we need to ACK what peer asked us for, not what we currently expect from it.